plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -327,6 +355,62 @@ func (s *SQLServer) accRow(query Query, acc telegraf.Accumulator, row scanner) e
 	return nil
 }
 
+// gatherHealth stores info about any query errors in the healthMetrics map
+func (s *SQLServer) gatherHealth(healthMetrics map[string]*HealthMetric, serv string, queryError error, mutex *sync.Mutex) {
+	if !s.HealthMetric {
+		return
+	}
+
+	mutex.Lock()
```

Comment:
publishing the time spent on this block might be helpful to indicate it is light weight and fast

Reply:
Sure, good call. For others: I timed the code block from right before `Lock()` until right after `Unlock()` it was consistently taking 1s of nanoseconds to `Lock`, execute the codeblock, then `Unlock`.

Reply:
I feel more safe if you put a `defer mutex.Unlock()` right below that line.

Reply:
Btw: If this is really a discussion point about holding the log for that few operations, just change the pattern to return if the query succeeded or not (i.e 1 if successful 0 otherwise). Then in the caller hold the lock and sum up the successful ones in `SuccessfulQueries` and add one to `AttemptedQueries` in the `healthMetrics` map.

In general I get the feeling that this is a very complicated way to count. :-)